OpenSim/Common/Component.cpp

```diff
@@ -1035,6 +1033,25 @@ void Component::setNextSubcomponentInSystem(const Component& sub) const
     }
 }
 
+void Component::updateFromXMLNode(SimTK::Xml::Element& node, int versionNumber)
+{
+    if (versionNumber < XMLDocument::getLatestVersion()) {
+        if (versionNumber < 30508) {
+            // Here's an example of the change this function might make:
+            // Previous: <connectors>
+            //               <Connector_PhysicalFrame_ name="parent">
+            //                   <connectee_name>...</connectee_name>
+            //               </Connector_PhysicalFrame_>
+            //           </connectors>
+            // New:      <connector_parent_connectee_name>...
+            //               </connector_parent_connectee_name>
+            //
```

Comment:
> which isn't totally clear about what is auto generated and what is user defined.

The comment is completely auto-generated. When the user gets a default XML representation for a component from the GUI's XML Browser (or from `opensim-cmd print-xml`), they will see:

```xml
<MyComponent>
  <!--Path to a Component to satisfy the Connector 'childFoo' of type Foo (description: The frame to which this geometry is attached. Note, being connected to a Frame means its transform is used to position this Geometry.).--> 
  <connector_frame_connectee_name></connector_frame_connecteee_name>
</MyComponent>
```

So the only thing that is user-defined is the value of the `<connector_frame_connectee_name>` element.


> but if the comments disappear (e.g. MATLAB XML editing) then determining the type that satisfies "parent" can be challenging.

It may be possible to make the field name contain the type: `<connector_PhysicalFrame_frame_connectee_name>`. I'm not sure if this is possible to implement, and might look bad for templated types (e.g., `<connector_TableSource_double__frame_connectee_name>`.

While I understand the sentiment of more directly putting the type name in the XML tags, I'm not sure it's the right thing to do. The type of the Connector is indeed documentation, not user-editable. So I think it is a benefit that the type name does not appear "user-editable" in the XML file. For example, users might mis-specify a Connector to Body with the erroneous tag `<Connector_Ground_ name="frame">`. 

I think it is reasonable to expect that the comments are necessary for users to know how to fill in the XML fields. I think we already hold this expectation for our XML files. For most existing properties, I would not feel comfortable editing them without seeing the XML comment (i.e., Body's `<inertia>`, Muscle's `<max_contraction_velocity>`). Also, the workflows that involve manually editing XML files should hopefully be the same workflows that preserve XML comments (namely, I'm thinking of using Models/setup files in the distribution's Models folder, and the GUI's XML Browser). Surely, if one edits an XML file in MATLAB using the OpenSim API and calls `Object::print()`, they will still see XML comments. And we want to discourage using XML libraries directly to edit the files.

We can also ensure that the error message obtained when specifying a path to an object of the wrong type is very clear. For example, "The component at path '/foo/bar' is of type Ground, but must be of type Body." (I didn't check the code yet to see if such an error message is already in place).